lib/ansible/plugins/lookup/template.py

```diff
@@ -87,8 +87,8 @@ def run(self, terms, variables, **kwargs):
                             newsearchpath.append(os.path.join(p, 'templates'))
                             newsearchpath.append(p)
                         searchpath = newsearchpath
-                    else:
-                        searchpath = [self._loader._basedir, os.path.dirname(lookupfile)]
+                    searchpath.append(os.path.dirname(lookupfile))
```

Comment:
this should be prepended, as the list controls the priority in which the search happens, the template itself should be the first location scanned